### PR TITLE
read SDRF file from a shell environment variable

### PR DESCRIPTION
**Pull request by pmb59**
The method summariseAtlasExperiment in functions.R was modified to read SDRF file from a shell environment variable , if available

### CODE REVIEW

File: R/functions.R [hunk 1]
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
 		stop( paste( "XML config file", atlasExperimentXMLfile, "does not exist or is not readable." ) )
 	}
     
-    cat( paste( "Reading experiment config from", atlasExperimentXMLfile, "...\n" ) )
+  cat( paste( "Reading experiment config from", atlasExperimentXMLfile, "...\n" ) )
```
Comment (pinin4fjords) [inline R/functions.R]: Could you fix the indentation levels please? The existing code is terrible for it as well, this is our chance to fix it! I mean throughout this routine, not just the highlighted line.

File: R/functions.R [hunk 2]
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
...
+  cat( "Finished reading experiment config\n" )
 
-	# Get the pipeline code from the experiment accession e.g. MTAB, MEXP
-	pipeline <- gsub( "E-", "", experimentAccession )
-	pipeline <- gsub( "-\\d+", "", pipeline )
-
-	# Filename for SDRF.
-	sdrfBasename <- paste( experimentAccession, ".sdrf.txt", sep="" )
+  #read SDRF file from a shell environment variable if it exists (otherwise it returns "")
+  sdrfPath <- Sys.getenv(x = 'SDRF_PATH', unset = "", names = NA)
```
Comment (pinin4fjords) [inline R/functions.R]: NA is probably a better solution than an empty string, then test with is.na() on the following line.

Comment (pcm32) [inline R/functions.R]: We can drop the x here
```suggestion
  sdrfPath <- Sys.getenv('SDRF_PATH', unset = "", names = NA)
```

File: R/functions.R [hunk 3]
```diff
@@ -127,39 +127,47 @@ summarizeAtlasExperiment <- function( experimentAccession, atlasExperimentDirect
...
+    
+  # if SDRF path not found
+  if (sdrfPath == ""){
+      
+    # Get the pipeline code from the experiment accession e.g. MTAB, MEXP
+	  pipeline <- gsub( "E-", "", experimentAccession )
+	  pipeline <- gsub( "-\\d+", "", pipeline )
```
Comment (pinin4fjords) [inline R/functions.R]:
```suggestion
	  pipeline <- unlist(strsplit(experimentAccession, '-'))[2]
```

**Review by pcm32 — COMMENTED** [review]
LGTM in general. Could you add a small comment on the fairly non-compliant documentation that this function has and also on the documentation of summarizeAtlasExperiment, explaining which env var needs to be set and what is the behaviour with or without this env var being set. Thanks.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]

File: R/functions.R (hunk 1 above)
Comment (pmb59) [reply R/functions.R]: yes, we noticed that yesterday

File: man/summarizeAtlasExperiment.Rd [hunk 4]
```diff
@@ -15,7 +15,8 @@
 	}
 	\item{atlasExperimentDirectory}{
 	Path to the directory containing the Atlas XML config file and expressions
-	matrices
+	matrices. Optionally, a shell environment variable \code{SDRF_PATH} 
+	can be defined containing the path to the SDRF file
```
Comment (pcm32) [inline man/summarizeAtlasExperiment.Rd]: Great, this should do documentation wise.

**Review by pcm32 — APPROVED** [review]
On my side it looks good, and I see some indentation changes, but I haven't checked if this looks good on an IDE. Lets see what @pinin4fjords says.

**Comment by pinin4fjords** [comment]
Indentation still all over the place- I'll fix it

**Comment by pinin4fjords** [comment]
This file was full of mixed tabs and spaces for indentation which made quite illegible for me. This is due to the bad habits of one of our alumni, but @pmb59 make sure you've set your editor to use 4 spaces when you use the tab key.

File: R/functions.R [hunk 5]
```diff
@@ -94,129 +94,137 @@ createAtlasSiteConfig <- function( ) {
...
-        	sdrfPath <- file.path( aeSiteConfig$AE2_LOAD_DIR, "EXPERIMENT", pipeline, experimentAccession, sdrfBasename )
-    	}
-    	
-	} else if (pipeline == "ENAD") {
-    	sdrfPath <- file.path( aeSiteConfig$ENA_SUBMISSIONS_TARGET, pipeline, experimentAccession, sdrfBasename )
+    #read SDRF file from a shell environment variable if it exists (otherwise it returns "")
+    sdrfPath <- Sys.getenv( 'SDRF_PATH', unset = NA, names = NA )
+    
+    # if SDRF path not found
+    if (is.na(sdrfPath) == TRUE){
```
Comment (pinin4fjords) [inline R/functions.R]: You don't need the == TRUE here

**Review by pinin4fjords — APPROVED** [review]

**Comment by pcm32** [comment]
For the record, I have checked independently (mamba installed previous version + devtools + optparse, then used install_github from R on that conda env and run https://github.com/ebi-gene-expression-group/bulk-recalculations/blob/develop/bin/createAtlasExperimentSummary.R) and it is working well:

```
(test-sdrf-path) bash-4.4$ Rscript script.R --source E-MTAB-5577 --accession E-MTAB-5577 --output my.Rdata
Reading experiment config from E-MTAB-5577/E-MTAB-5577-configuration.xml ...
Finished reading experiment config
Reading SDRF from /myPath/EXPERIMENT/MTAB/E-MTAB-5577/E-MTAB-5577.sdrf.txt ...
Getting characteristic and factor column indices
Checking for technical replicates
Locating assay names...
Found assay names. Subsetting SDRF...
Finished subsetting SDRF.
No Unit columns found.
Fixing column headings...
Finished fixing column headings.
Removing duplicated columns...
Finished removing duplicated columns.
Applying new column headings...
Finished applying new column headings.
Removing duplicated rows...
Finished removing duplicated rows.
Finished reading SDRF
Creating experiment summary...
Experiment summary created.
(test-sdrf-path) bash-4.4$ env | grep 'SDRF_PATH'
SDRF_PATH=/myPath/EXPERIMENT/MTAB/E-MTAB-5577/E-MTAB-5577.sdrf.txt
```

with a copy of E-MTAB-5577 from analysis/differential/rna-seq.

**Pull request merged**